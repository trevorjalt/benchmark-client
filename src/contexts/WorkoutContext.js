import React, { Component } from 'react'

const WorkoutContext = React.createContext({
    workoutList: [],
    exerciseList: [],
    exerciseSetList: [],
    newWorkoutItem: {},
    error: null,
    isLoggedIn: null,
    clearError: () => {},
    setError: () => {},
    setIsLoggedIn: () => {},
    setWorkoutList: () => {},
    setExerciseList: () => {},
    onDeleteExercise: () => {},
    onDeleteExerciseSet: () => {},
    onDeleteWorkout: () => {},
    onUpdateExercise: () => {},
    onUpdateExerciseSet: () => {},
})
export default WorkoutContext

export class WorkoutProvider extends Component {
    state = {
        workoutList: [],
        exerciseList: [],
        exerciseSetList: [],
        newWorkoutItem: {},
        error: null,
        isLoggedIn: null,
    }

    clearError = () => {
        this.setState({ error: null })
    }

    setError = error => {
        console.error(error)
        this.setState({ error })
    }

    setIsLoggedIn = (value) => {
        this.setState({ isLoggedIn: value })
    }

    setWorkoutList = workoutList => {
        this.setState({ workoutList})
    }

    setExerciseList = exerciseList => {
        this.setState({ exerciseList})
    }

    setExerciseSetList = exerciseSetList  => {
        this.setState({ exerciseSetList })
    }

    onDeleteExercise = (newList) => {
        this.setExerciseList(newList)
    }

    onDeleteExerciseSet = (newList) => {
        this.setExerciseSetList(newList)
    }

    onDeleteWorkout = (newList) => {
        this.setWorkoutList(newList)
    }

    onUpdateExercise = (newList) => {
        this.setExerciseList(newList)
    }

    onUpdateExerciseSet = (newList) => {
        this.setExerciseSetList(newList)
    }

    render() {
        const value = {
            workoutList: this.state.workoutList,
            exerciseList: this.state.exerciseList,
            exerciseSetList: this.state.exerciseSetList,
            newWorkoutItem: this.state.newWorkoutItem,
            error: this.state.error,
            isLoggedIn: this.state.isLoggedIn,
            touched: this.state.touched,
            clearError: this.clearError,
            setError: this.setError,
            setIsLoggedIn: this.setIsLoggedIn,
            setWorkoutList: this.setWorkoutList,
            setExerciseList: this.setExerciseList, 
            setExerciseSetList: this.setExerciseSetList,
            onDeleteExercise: this.onDeleteExercise,
            onDeleteExerciseSet: this.onDeleteExerciseSet,
            onDeleteWorkout: this.onDeleteWorkout,
            onUpdateExercise: this.onUpdateExercise,
            onUpdateExerciseSet: this.onUpdateExerciseSet,         
        }
        return (
            <WorkoutContext.Provider value={value}>
                {this.props.children}
            </WorkoutContext.Provider>
        )
    }
}