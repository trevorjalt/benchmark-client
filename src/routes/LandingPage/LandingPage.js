import React, { Component } from 'react'
import Landing from '../../components/Landing/Landing' 
import RegistrationForm from '../../components/RegistrationForm/RegistrationForm'


export default class LandingPage extends Component {
    static defaultProps = {
        history: {
            push: () => {}
        },
    }

    handleRegistrationSuccess = user => {
        const { history } = this.props
        history.push('/login')
    }
    render() {
        return (
            <section className='LandingPage'>
                <Landing />
                <RegistrationForm 
                    onRegistrationSuccess={this.handleRegistrationSuccess}
                />
            </section>
        )
    }
}