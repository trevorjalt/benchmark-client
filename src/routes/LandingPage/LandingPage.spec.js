import React from 'react'
import { WorkoutProvider } from '../../contexts/WorkoutContext'
import LandingPage from './LandingPage'
import toJson from 'enzyme-to-json'
import { shallow } from 'enzyme'

describe(`LandingPage Route`, () => {
    it (`renders without crashing`, () => {
        const wrapper = 
            shallow (
                <WorkoutProvider>
                    <LandingPage />
                </WorkoutProvider>
            )
        expect(toJson(wrapper)).toMatchSnapshot()
    })
})