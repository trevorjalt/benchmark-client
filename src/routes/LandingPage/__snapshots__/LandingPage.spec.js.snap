// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LandingPage Route renders without crashing 1`] = `
<ContextProvider
  value={
    Object {
      "clearError": [Function],
      "error": null,
      "exerciseList": Array [],
      "exerciseSetList": Array [],
      "isLoggedIn": null,
      "newWorkoutItem": Object {},
      "onDeleteExercise": [Function],
      "onDeleteExerciseSet": [Function],
      "onDeleteWorkout": [Function],
      "onUpdateExercise": [Function],
      "onUpdateExerciseSet": [Function],
      "setError": [Function],
      "setExerciseList": [Function],
      "setExerciseSetList": [Function],
      "setIsLoggedIn": [Function],
      "setWorkoutList": [Function],
      "touched": undefined,
      "workoutList": Array [],
    }
  }
>
  <LandingPage
    history={
      Object {
        "push": [Function],
      }
    }
  />
</ContextProvider>
`;
